src/app/services/role-guard.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Router } from '@angular/router';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RoleGuardService {
+  constructor(private auth: AuthService, private router: Router) { }
+
+  // TODO: instead of redirecting to login, show a message to the user
+  canActivate(route: ActivatedRouteSnapshot) {
+    if (!this.auth.isAuthenticated()) {
+      this.router.navigate(['login']);
+      return false;
+    }
+
+    const token = AuthService.getAccessToken();
+
+    if (token == null) {
+      this.router.navigate(['login']);
+      return false;
+    }
+    
+    const roles = route.data['roles'] as Array<string>; // TODO string enum for roles
```

Comment:
We should have an enum to take these roles from, preferably standardized with the database.